src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalesceAssignmentTests.cs

```diff
@@ -314,6 +314,44 @@ void M()
             Assert.True(assignmentTypeInfo.ConvertedType.IsErrorType());
         }
 
+        [Theory]
+        [InlineData("=")]
+        [InlineData("??=")]
+        public void ImplicitObjectCreation_UnconstrainedGenericType(string assignmentOperator)
+        {
+            var source = $$"""
+                class C
+                {
+                    void M<T>() where T : new()
+                    {
+                        T t = default;
+                        t {{assignmentOperator}} new();
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var descendantNodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var node = descendantNodes.OfType<ImplicitObjectCreationExpressionSyntax>().Single();
+
+            var typeInfo = model.GetTypeInfo(node);
+            Assert.Equal("T", typeInfo.Type.ToTestDisplayString());
+            Assert.Equal("T", typeInfo.ConvertedType.ToTestDisplayString());
+
+            var symbolInfo = model.GetSymbolInfo(node);
+            Assert.True(symbolInfo.IsEmpty);
```

Comment:
This frankly seems like a bug. @CyrusNajmabadi, is this the behavior you'd expect from this test?

Reply:
i agree.  that seems odd.  what's the reason for that?

Reply:
if we had `new T()` instead, do we get the same result?

Reply:
Oh, I see what's going on here. This is the generic `where T : new()` case. There literally isn't a constructor symbol here.

Reply:
@DoctorKrolic, I believe that explains what you were seeing: yes, it is expected that `where T : new()` won't show a symbol here, because there isn't one _to_ show.

Reply:
> it is expected that `where T : new()` won't show a symbol here, because there isn't one _to_ show.

Interesting. Given that `new()` implies, that type has a parameterless constructor, I thought compiler makes a constructor symbol for such type parameters.

With that question resolved, we don't expect anything to be returned in an equivalent error case then, right?

Reply:
> we don't expect anything to be returned in an equivalent error case then, right?

Correct, nothing should be returned. That being said, if there are some particularly odd cases that do, imo it's fine. Error scenarios are best effort.

Reply:
> There literally isn't a constructor symbol here

Out of curiosity: why is that?

> Given that `new()` implies, that type has a parameterless constructor, I thought compiler makes a constructor symbol for such type parameters.

Reply:
> I thought compiler makes a constructor symbol for such type parameters.

It doesn't. What metadata symbol would that reflect?